### PR TITLE
Move directory api to commons ui

### PR DESCRIPTION
**Pull request by Meklo**
depends on gridsuite/commons-ui#460

### CODE REVIEW

File: src/components/dialogs/commons/utils.tsx [hunk 1]
```diff
@@ -5,10 +5,10 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
-export function getIdOrValue(value: { id: string; label: string } | string) {
+export function getIdOrValue(value: any) {
     return typeof value !== 'string' ? value?.id ?? null : value;
 }
 
-export function getLabelOrValue(value: { id: string; label: string } | string) {
+export function getLabelOrValue(value: any) {
```
Comment (Tristan-WorkGH) [inline src/components/dialogs/commons/utils.tsx:12]: why did you remove typescript check?

**Review by flomillot — APPROVED** [review]

**Review by Tristan-WorkGH — COMMENTED** [review]
LGTM

Missing update of `@gridsuite/commons-ui` in `package.json` dependencies.

File: src/components/dialogs/commons/utils.tsx (hunk 1 above)
Comment (Meklo) [reply src/components/dialogs/commons/utils.tsx:12]: Compilation errors, too hard to properly type, seen with @flomillot

**Review by Tristan-WorkGH — APPROVED** [review]
okay, will fix the `any` in another PR
